poleno/utils/tests/test_templatetags.py

```diff
@@ -199,18 +201,32 @@ def language_view(request):
             u'request': request,
         })))
 
-    urls = tuple(patterns(u'',
+    urlpatterns = tuple(patterns(u'',
         url(r'^$', active_view, name=u'index'),
         url(r'^first/', active_view, name=u'first'),
         url(r'^second/', include(namespace=u'second', arg=patterns(u'',
             url(r'^$', active_view, name=u'index'),
             url(r'^first/', active_view, name=u'first'),
         ))),
     ))
-    urls += tuple(i18n_patterns(u'',
+    urlpatterns += tuple(i18n_patterns(u'',
         url(r'^language/', language_view, name=u'language'),
     ))
 
+    @staticmethod
+    def page_not_found(request):
+        template = Template(
+            u'{% load change_lang from poleno.utils %}'
+            u'{% change_lang lang.0 %}'
+        )
+        body = template.render(RequestContext(request, {u'request_path': request.path}))
+        return http.HttpResponseNotFound(body, content_type=u'text/html')
+    handler404 = page_not_found
+
+    @cached_property
+    def urls(self):
```

Comment:
@martinmacko47 Je to tak trocha hack. Pozeral som si, ako je napisany `TestCase` ktory pozaduje ako nazov `urls` pre "URLconf". Pricom akceptuje nazov modulu alebo pole s urls. 
```
        # TestCase
        if hasattr(self, 'urls'):
            self._old_root_urlconf = settings.ROOT_URLCONF
            settings.ROOT_URLCONF = self.urls
            clear_url_caches()
```
Pri definovani custom error view je potrebne aby handler bol zadefinovany v mieste ako "URLconf". To bol vsak problem kedze som do pola s urls nijako nemohol definovat handler. Dalsie spracovanie nastava v `django.core.urlresolver.RegexURLResolver` ktory vyzaduje nazov pre "URLconf" `urlpatterns` (co sa bilo s TestCase).
```
# RegexURLResolvers
patterns = getattr(self.urlconf_module, "urlpatterns", self.urlconf_module)
```
Sice `self` nevracia nazov modulu "URLconf" ale nazov classy. Vyzera ze to funguje dobre. Aj v [dokumentacii](https://docs.djangoproject.com/en/3.1/topics/http/views/#testing-custom-error-views) su `urls` a `handler*` definovany mimo class, ale, to by bolo nepriehladne.

Reply:
@viliambalaz s tym `return self` je to vazne riadny hack. Naozaj treba cely objekt so vsetkym nestaci nieco taketo?

```
    urls = Bunch(
        urlpatterns=urlpatterns,
        handler404=page_not_found,
    )
```

Akurat s tym rozdielom, ze `page_not_found` nesmie byt `@staticmethod`, ale obycajna funkcia ako je napriklad `language_view` vyssie:

```
    def page_not_found(request):
        ...
```

Reply:
:ok: ano funguje